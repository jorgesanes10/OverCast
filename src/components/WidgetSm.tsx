import { ReactNode } from 'react';
import styled from 'styled-components';

interface WidgetSmProps {
  label: string;
  value: ReactNode;
  icon: ReactNode;
  ariaLabel: string;
}

export default function WidgetSm({
  label,
  value,
  icon,
  ariaLabel,
}: WidgetSmProps) {
  return (
    <StyledWidgetSm aria-label={ariaLabel} className="widget">
      <div aria-hidden>
        <p>
          {icon} {label}
        </p>
        <p className="value">{value}</p>
      </div>
    </StyledWidgetSm>
  );
}

const StyledWidgetSm = styled.div`
  height: 100%;
  display: flex;

  div {
    text-align: left;
    display: flex;
    flex-direction: column;
    padding: 12px;
    color: #fff;
    background-color: #ffffff30;
    border-radius: 8px;
    flex-grow: 1;
  }

  p {
    margin: 0;
    font-weight: 100;
  }

  p.value {
    margin: 24px 0;
    font-size: 28px;
    align-self: center;
    font-weight: 400;
  }

  svg {
    position: relative;
    top: 5px;
  }
`;
