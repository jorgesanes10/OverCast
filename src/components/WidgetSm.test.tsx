import { render, screen } from '@testing-library/react';
import WidgetSm from './WidgetSm.tsx';

describe('WidgetSm', () => {
  test('renders the component', () => {
    const { container } = render(
      <WidgetSm
        label="My label"
        value="My value"
        icon="My icon"
        ariaLabel="My aria label"
      />,
    );

    expect(screen.getByText('My icon My label')).toBeDefined();
    expect(screen.getByText('My value')).toBeDefined();

    expect(container).toMatchSnapshot();
  });
});
