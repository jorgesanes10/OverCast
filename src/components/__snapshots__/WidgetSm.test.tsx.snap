// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetSm renders the component 1`] = `
<div>
  <div
    aria-label="My aria label"
    class="sc-KXCwU icEFEp widget"
  >
    <div
      aria-hidden="true"
    >
      <p>
        My icon
         
        My label
      </p>
      <p
        class="value"
      >
        My value
      </p>
    </div>
  </div>
</div>
`;
